### DISCUSSION TITLE
Home Assistant Container LXC - Internet NOT Connected issues

### DISCUSSION THREAD

**Discussion by thlucas1** [tteck/Proxmox#2006] [Help Needed] [answered]
Hi All,

I am trying to setup a Proxmox Home Assistant Container LXC to help a user troubleshoot a problem with an HA integrations that I wrote.  I have Proxmox 8.0.4 installed, and am able to configure things via the GUI just fine.  I can ping machines on my local network as well as the internet via the main node (proxmox01) console just fine.  I know that the preferred way of installing this is a VM, but I wanted to install it the way the user has it installed (Container LXC) to more closely mirror his environment (e.g. VM install is not an option).

I used the following `Home Assistant Container LXC` link from the [Proxmox VE Helper Scripts](https://tteck.github.io/Proxmox/) page, entering the following command in the proxmox01 node console:
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/homeassistant.sh)"`

I select "Advanced" setup type, and go through the questions.  I named the new container `homeassistantct101`, and gave is a static IP of `192.168.1.112`, 4GB RAM, 32GB HDD, and selected Verbose mode.

I get to the point in the script where it gives me a "Network Connected: 192.198.1.112" message (which is good), but then fails with a "Internet NOT Connected" message immediately after that.

Here is the output of the script just before it fails:
```
...
 ✓ Set up Container OS
++ hostname -I
+ msg_ok 'Network Connected: \033[36m192.168.1.112 '
+ local 'msg=Network Connected: \033[36m192.168.1.112 '
+ echo -e '\r\033[K \033[1;92m✓\033[m \033[1;92mNetwork Connected: \033[36m192.168.1.112 \033[m'
 ✓ Network Connected: 192.168.1.112 
+ network_check
+ set +e
+ trap - ERR
+ ping -c 1 -W 1 1.1.1.1
+ msg_error 'Internet NOT Connected'
+ local 'msg=Internet NOT Connected'
+ echo -e '\r\033[K \033[01;31m✗\033[m \033[01;31mInternet NOT Connected\033[m'
 ✗ Internet NOT Connected
+ read -r -p 'Would you like to continue anyway? <y/N> ' prompt
Would you like to continue anyway? <y/N> 
```

At this point, I select `N` to stop the script, which returns me to proxmox01 console.  I see that it is trying to ping the CloudFare DNS server at 1.1.1.1 to determine the Internet Connection status.  I can issue the `ping -c 1 -W 1 1.1.1.1` command from the proxmox01 console and it returns fine.  I then switch over to the new `homeassistantct101` container and issue the `ping 1.1.1.1` command, which fails (the `hostname -I` command returns ok):
```
root@homeassistantct101:~# hostname -I
192.168.1.112 
root@homeassistantct101:~# ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
From 192.168.1.112 icmp_seq=1 Destination Host Unreachable
From 192.168.1.112 icmp_seq=2 Destination Host Unreachable
From 192.168.1.112 icmp_seq=3 Destination Host Unreachable
From 192.168.1.112 icmp_seq=4 Destination Host Unreachable
From 192.168.1.112 icmp_seq=5 Destination Host Unreachable
From 192.168.1.112 icmp_seq=6 Destination Host Unreachable
From 192.168.1.112 icmp_seq=7 Destination Host Unreachable
^C
--- 1.1.1.1 ping statistics ---
7 packets transmitted, 0 received, +7 errors, 100% packet loss, time 6138ms
pipe 4
```

For some reason, the container cannot see the external internet.

What am I doing wrong?

Thanks - Todd

**Comment by tteck** [comment]
>gave is a static IP of 192.168.1.112

It needs to be a proper IPv4 CIDR Address `192.168.1.112/24`

**Reply by tteck** [reply]
If that was entered correctly, then it's DNS

**Reply by tteck** [reply]
Try utilizing Default settings to ensure that mistakes weren't made with Advanced settings.

**Reply by thlucas1** [reply]
I did enter it as an IPv4 CIDR "192.168.1.112/24" in the Advanced settings.  I've tried tweaking the settings about 20 times now.  Using default settings yields the same results.  I also tried specifying MAC address (and not), as well as different DNS settings (default to host, using a specific DNS server, etc), disabling / enabling IPv6, etc.   The closest I get to success is what I described, but it fails on the "Internet NOT Connected".

Is there something I need to set at the host proxmox01 node level maybe?  e.g. some network settings.  Should I specify the host proxmox01 node IPv4 address as the gateway server instead of my router address?  

I should also mention that I am running the Proxmox server instance as a VM node in my VMWare ESXI 6.7 server.  I don't think that would make a difference, but not sure.  I would say that it is fine since I can ping local and internet addresses from the proxmox01 node console just fine, which tells me it's flowing through the VMWare networking environment ok.

Any other ideas?  I appreciate your help and the quick response - thank you!

**Reply by thlucas1** [reply]
Forgot to mention I also tried both Ubuntu and Debian operating systems options (as well as Debian 11 and 12), and also tried the unprivileged versus privileged user.

**Reply by tteck** [reply]
>I should also mention that I am running the Proxmox server instance as a VM node in my VMWare ESXI 6.7 server

I doubt that setup will work. Good luck.

**Reply by thlucas1** [reply]
The fact that I can ping internal network and external network addresses from the proxmox01 node tells me it's working fine at that level.  It's just when I try to ping internal or external addresses from a container that it is blocked.  Firewall is turned off at the datacenter level, so it rules that out.

Here's my network settings for the `homeassistantct101` container:
![image](https://github.com/tteck/Proxmox/assets/38439576/bd935a11-8aa4-4b28-a4d6-9eaa4c6e806c)

Here's my network settings for the parent `proxmox01` node:
![image](https://github.com/tteck/Proxmox/assets/38439576/4a5fd5a8-d02f-442b-98a4-4c886b608e0d)

**Reply by thlucas1** [reply, marked as answer]
nevermind, I figured it out.

I had to enable promiscuous mode and forged transmits on my ESXi vswitch.

Thanks for your help.

**Reply by mikymontero** [reply]
I don't understand how you fixed it and I'm the same as you

**Reply by thlucas1** [reply]
I enabled promiscuous mode and forged transmits on my ESXi vswitch security settings:
![image](https://github.com/user-attachments/assets/8e002454-2c7b-4d09-8720-57ee2e4f4aa9)

and here's the VM settings:
![image](https://github.com/user-attachments/assets/ebd205ef-5927-4254-99d0-46d011f95e3b)

Hope it helps

**Reply by mikymontero** [reply]
where do i find this

**Reply by thlucas1** [reply]
I run vSphere Hypervisor 6.7 (e.g. ESXi 6.7), which has its own web user-interface.  Here is what mine looks like:

![image](https://github.com/user-attachments/assets/0f63dfab-6027-4277-9f91-3110561eac3e)

**Reply by mikymontero** [reply]
I'll look to see if I can find it Thx
